apps/meteor/app/lib/server/lib/getHiddenSystemMessages.ts

```diff
@@ -1,26 +1,8 @@
 import type { MessageTypesValues, IRoom } from '@rocket.chat/core-typings';
 
-import { settings } from '../../../settings/server';
+import { getCachedHiddenSystemMessage } from '../../../../server/lib/systemMessage/hideSystemMessage';
 
-const hideMessagesOfTypeServer = new Set<MessageTypesValues>();
-
-settings.watch<MessageTypesValues[]>('Hide_System_Messages', (values) => {
-	if (!values || !Array.isArray(values)) {
-		return;
-	}
-
-	const hiddenTypes = values.reduce((array, value): MessageTypesValues[] => {
-		const newValue: MessageTypesValues[] = value === 'mute_unmute' ? ['user-muted', 'user-unmuted'] : [value];
-
-		return [...array, ...newValue];
-	}, [] as MessageTypesValues[]);
-
-	hideMessagesOfTypeServer.clear();
-
-	hiddenTypes.forEach((item) => hideMessagesOfTypeServer.add(item));
-});
-
-// TODO probably remove on chained event system
-export function getHiddenSystemMessages(room: IRoom): MessageTypesValues[] {
-	return Array.isArray(room?.sysMes) ? room.sysMes : [...hideMessagesOfTypeServer];
-}
+export const getHiddenSystemMessages = async (room: IRoom): Promise<MessageTypesValues[]> => {
```

Comment:
it's better to pass the setting value as a parameter, so the other places can use the normal settings.get while this one uses the call to the model